core/chainio/avs_writer.go

```diff
@@ -65,15 +66,16 @@ func BuildAvsWriter(txMgr txmgr.TxManager, registryCoordinatorAddr, operatorStat
 	if err != nil {
 		return nil, err
 	}
-	return NewAvsWriter(avsRegistryWriter, avsServiceBindings, logger, txMgr), nil
+	return NewAvsWriter(avsRegistryWriter, avsServiceBindings, ethHttpClient, logger, txMgr), nil
 }
 
-func NewAvsWriter(avsRegistryWriter avsregistry.AvsRegistryWriter, avsServiceBindings *AvsManagersBindings, logger logging.Logger, txMgr txmgr.TxManager) *AvsWriter {
+func NewAvsWriter(avsRegistryWriter avsregistry.AvsRegistryWriter, avsServiceBindings *AvsManagersBindings, client eth.Client, logger logging.Logger, txMgr txmgr.TxManager) *AvsWriter {
 	return &AvsWriter{
 		AvsRegistryWriter:   avsRegistryWriter,
 		AvsContractBindings: avsServiceBindings,
-		logger:              logger,
 		TxMgr:               txMgr,
+		client:              client,
```

Comment:
Feels like an excessive member just for on function

Reply:
It was already there, just never initialized

Reply:
Removed it in [4629ef4](https://github.com/NethermindEth/near-sffl/pull/136/commits/4629ef418d7ba2ac1f3852a0cc31bad62d0898ee)

Reply:
> It was already there, just never initialized

It was passed as argument and propagated to struct that actually needs it to make txs. We just use those structs

Reply:
Yeah, what I mean is it was a field of `AvsWriter`